### DISCUSSION TITLE
What import attribute types are available?

### DISCUSSION THREAD

**Discussion by justinmchase** [denoland/deno#23696] [Q&A] [answered]
I see this is working in Deno:

```ts
import example from "./example.json" with { type: "json" };
```

Thats great!

But I am wondering what types are available other than `"json"`? The intellisense in vscode just says `any` for the whole with block and I can't seem to find the docs, which all seem to focus heavily on just importing json.

Also, this is bit of a leading question because specifically what I'm looking for is the ability to define _custom_ types and handle those as a sort of pre-processor.

#### some hypothetical syntax
```ts
import justin "./justin.ts"
import example from "./example.justin" with justin;
```

I really have no idea if the above is possible, or if anyone is even thinking along these lines as an _eventually_ feature? 

But I'm not gonna lie it would be pretty cool. I'm working on a tool which is essentially a parser generator and the above would be pretty slick for people making dsls. 

#### json alternative (simplified)
```ts
import justin from "./justin.ts"
import dsl from "./example.json" with { type: "json" };
export const example = justin(dsl);
```

The above will work for now but it will essentially require a pre-compile step for dsl authors while the desired _hypothetical_ syntax would be dynamic, which feels more appropriate for the web.

I'm just wondering how far this feature will go.

**Comment by bartlomieju** [comment, marked as answer]
> But I am wondering what types are available other than "json"? The intellisense in vscode just says any for the whole with block and I can't seem to find the docs, which all seem to focus heavily on just importing json.

Currently only `"json"` type is supported. We did a bunch of preliminary work to support `text`, `bytes` and `wasm` types as well (eg.  https://github.com/denoland/deno_core/pull/402) but they are not yet available.

> I really have no idea if the above is possible, or if anyone is even thinking along these lines as an eventually feature?

There was talks about this in the team, especially @piscisaureus wanted such a feature, but so far there's been no consensus and we don't plan doing that in the forseeable future.

**Reply by justinmchase** [reply]
Thank you for the update, I am pretty happy with the json version at the moment so that should keep be busy for a while.

**Comment by Symbitic** [comment]
@bartlomieju any updates on this? I'm depending on `text` imports for a framework I'm working on.

**Reply by marvinhagemeister** [reply]
No updates so far.
